app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -396,14 +396,13 @@ function FormSteps({
     }
 
     onStepSubmit(step?.name);
-
     const nextStepIndex = stepIndex + 1;
     const isComplete =
       stepCanComplete !== undefined ? stepCanComplete : nextStepIndex === steps.length;
     if (isComplete) {
       onComplete(values);
     } else {
-      const { name: nextStepName } = steps[nextStepIndex];
+      const { name: nextStepName } = steps[nextStepIndex] ?? steps[steps.length - 1];
```

Comment:
Can you explain why this is needed? My impression would be that either the current step is meant to complete the flow (`stepCanComplete`) or it's the last step in the sequence, so I'm not clear why we would look at the penultimate step.

I think a test case would be useful to describe the context and expected behavior.

Reply:
So my understanding of the React fsm is pretty minimal, but basically what was happening is that when the pages are swapped in order (i.e. prepare before location search) we get an error here because name cannot be destructured from the steps array at the nextStepIndex. It was returning undefined. 

This is a bit hacky to avoid that error. The goal is to just set the index to the last step if its null or undefined.

Reply:
If I recall correctly, IPP was doing something unique with skipping steps to completion, which is what `stepCanComplete` was added for. Was this happening in the previous "last" step (prepare) and just needs to now happen in the new last step (location) instead?

Reply:
Yeah, that's a good question. I tried to use `stepCanComplete`, but had some difficulty getting it to do what I wanted at the time. 

I'm gonna go back a dig around some more to see I can better understand what's happening.

Reply:
So in digging around a bit more, I realized we can delete the entire else statement. The changes I made manually set the the next step name when the user clicks through the flow. So the nullish coalescing bit is uneccessary and the error itself was likely fixed before I added in this code.

Reply:
> I realized we can delete the entire else statement. 

That doesn't sound right to me, since the component is meant to manage a series of steps, and without this it wouldn't progress from one step to the next?

Reply:
I guess I'd defer to your team to decide how best to maintain this now that IPP is the primary consumer of "steps" in the document capture screen, but the intent of the component was originally to try to manage a lot of the concerns with moving through a series of steps based on form submission, and it seems like we're bypassing a lot of that value now with ad hoc step changes.

Above all, I guess I'm surprised that the component needed to change at all as a result of reordering steps? 🤷

Reply:
No I totally agree with your surprise, and am trying to suss out what's going on there. 

I'm of two minds about it, on the one hand there's really only two steps that we're handling here, so do we need all this additional machinery, on the other hand we should probably use what's already there.

Reply:
Following up here: 
I opted to have most of the steps continue to work with the flow state machine, with the exception being the location search step which exits the flow. 

As a consequence, most of the steps now use submit buttons, and for those steps that do we no longer prevent the default behavior of those buttons.

Reply:
I haven't had a chance to take a close look, but using the native submit behavior sounds like a good path if we can make that work 👍